internal/provider/instance/data_source.go

```diff
@@ -0,0 +1,138 @@
+package instance
```

Comment:
Would this package (and directory) be better named as `collectioninstance`? That's at least a little less ambiguous.

Reply:
Changed to `collection_instance` to match how I named the packages in managed workloads e.g. `agent_configuration`.